foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/mappings/OneToManyMapping.java

```diff
@@ -309,7 +309,12 @@ public void collectQueryParameters(Set<DatabaseField> cacheFields){
     @Override
     public Object clone() {
         OneToManyMapping clone = (OneToManyMapping)super.clone();
-        clone.setTargetForeignKeysToSourceKeys(new HashMap<>(getTargetForeignKeysToSourceKeys()));
+
+        Map<DatabaseField, DatabaseField> old2cloned = new HashMap<>();
+        clone.sourceKeyFields = cloneDatabaseFieldVector(sourceKeyFields, old2cloned);
+        clone.targetForeignKeyFields = cloneDatabaseFieldVector(targetForeignKeyFields, old2cloned);
+        clone.setTargetForeignKeysToSourceKeys(cloneKeysMap(getTargetForeignKeysToSourceKeys(), old2cloned));
+        clone.sourceKeysToTargetForeignKeys = cloneKeysMap(getSourceKeysToTargetForeignKeys(), old2cloned);
```

Comment:
one should be careful here. Based on the test being added, these two collections are not needed. For the sake of correctness, I believe it makes sense to clone them too, but make sure it's being done correctly, see the comment in [OneToOneMapping.clone()](https://github.com/eclipse-ee4j/eclipselink/blob/3.0.1/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/mappings/OneToOneMapping.java#L453)

Reply:
Do I need to do some additinal checks?